packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -823,10 +817,16 @@ export class ContainerRuntime
 		let idCompressor: (IIdCompressor & IIdCompressorCore) | undefined;
 		if (idCompressorEnabled) {
 			const { IdCompressor, createSessionId } = await import("./id-compressor");
-			idCompressor =
-				serializedIdCompressor !== undefined
-					? IdCompressor.deserialize(serializedIdCompressor, createSessionId())
-					: IdCompressor.create(logger);
+
+			const pendingLocalState = context.pendingLocalState as IPendingRuntimeState;
+
+			if (pendingLocalState?.pendingIdCompressorState !== undefined) {
+				idCompressor = IdCompressor.deserialize(pendingLocalState.pendingIdCompressorState);
```

Comment:
One tenet of this work thus far to integrate the compressor into the runtime has been to not change the compressor's existing APIs. 

With this in mind, when the compressor gets stashed, it has the compressor's state at "stash time" and not at summary time. If a saved op is replayed, the stashed compressor that we've rehydrated from will have already seen and finalized the Id, which will fail. 

It may technically be possible to change the compressor's deserialization to assume the session Id from the snapshot and "re-allocate" the Ids from the replayed ops, but we would need to be careful that the exact same Ids are spit out in the same order as otherwise anything utilizing the Ids would now have the incorrect Id. Let me know if you're really against doing it this way with metadata and I can do a more in-depth exploration (and ask people who would know better) if doing it this way is possible.